### ISSUE TITLE
Add Firefox ARM support

### ISSUE THREAD

**Issue by mossymossmoss** [SimonBrazell/privacy-redirect#354]
I use a PinePhone which is running the Arm build of Firefox and having Privacy Redirect would help towards a lighter web experience. I believe that Mozilla just refers to ARM support as Android support, so it may be that which is necessary to look up the documentation for

**Comment by mu88** [comment]
Just to make sure that I don't create a duplicate: are you requesting support for "Firefox for Android"? Because that's what I am looking for

**Comment by mossymossmoss** [comment]
Yes, that's essentially what this issue is for 👍

**Comment by sjehuda** [comment]
If there's a Greasemonkey extension for ARM, you can use this User.JS script https://openuserjs.org/scripts/sjehuda/Farside

**Comment by mu88** [comment]
Unfortunately, there seems to be no Greasemonkey add-on for Firefox for Android

**Comment by sjehuda** [comment]
What about Tampermonkey?
Is it available on Firefox for Android?

**Comment by mu88** [comment]
No :(

**Comment by sjehuda** [comment]
I though to myself that addons are architecture agnostic.
It's a good idea to bring Userscripts to ARM [web browser](https://search.f-droid.org/?q=user.js&lang=en).
I suggest to [start a thread](https://forum.f-droid.org/) at F-Droid.

**Comment by mu88** [comment]
Hmmm, I found this dedicated page [Developing extensions for Firefox for Android](https://extensionworkshop.com/documentation/develop/developing-extensions-for-firefox-for-android/) - so probably it's not agnostic.

And what should we ask F-Droid for?

**Comment by sjehuda** [comment]
I think F-Droid is a good hub for finding people who would want to promote good ideas, namely as to how to increase privacy, but also to promoting a Userscript manager.
For after all, Userscript manager is a program which let's you install an infinite range of other programs (in User.JS format), so I think this reason alone will suffice.

**Comment by sjehuda** [comment]
@mu88 did you try [UntrackMe](https://f-droid.org/en/packages/app.fedilab.nitterizeme/)?

**Comment by mu88** [comment]
Yes. That works okay when opening links from other apps (e. g. out of a messenger). But as soon as I'm already on a website in my default browser and clicking a YouTube link, that approach fails. I'd have to share the link with UntrackMe, which does the redirect and brings me back to my browser.

That would be the benefit of a dedicated browser add-on: no other app would be needed and it would work for both use cases (coming from an external app and coming from a website).

**Comment by sjehuda** [comment]
> That would be the benefit of a dedicated browser add-on: no other app
> would be needed and it would work for both use cases (coming from an
> external app and coming from a website). 
> 

I agree.

**Comment by sjehuda** [comment]
This might be relevant https://github.com/libredirect/libredirect/issues/75

**Comment by GreenLunar** [comment]
@mu88 according to https://greasyfork.org/en there are several Userscript managers for Android.  Install a userscript manager and then install [FREEdirector](https://openuserjs.org/scripts/sjehuda/FREEdirector). And if anything is missing, please let me know.

**Comment by mu88** [comment]
Okay, that works - thank you so far 👍🏻

Just for my understanding: do I understand all those issues/threads correctly that Firefox actually does not accept add-ons like Tampermonkey for Android being listed in their official add-on store and therefore these workarounds are necessary?

**Comment by sjehuda** [comment]
I'm not sure I understand.

If you specifically refer to FREEdirector, then the answer is that it's a Userscript (javascript script on user's end) which is managed by a userscript manager.

Userscript managers allow you to add javascript code that would change the way you interact with websites.

For example [Black Belt](https://openuserjs.org/scripts/sjehuda/Black_Belt) will display a black bar at top of a page, ones it finds Feeds, Torrents, PGP Keys, Apps, Cryptowallets, Magnet Links etc. etc.

Other userscripts may change the entire interface of a website.

By the nature Userscripts have began since pre-HTML5 era to this day, they won't be allowed by a multi-billion cooperation such as the one responsible for (today's) spyware web browser which (by my observation) regulates its affairs, including addons, namely for special interest and is biased towards advertizers who have no interest in users that have the power to change websites behaviour.

YOU SHOULD INSTALL SCRIPTS FROM SOURCES YOU TRUST!

**Comment by mu88** [comment]
I am sorry if I have expressed myself a little weird 😜 What I mean is the following:

On Firefox' desktop browser, all I have to do is install the PrivacyRedirect add-on from Firefox' official add-on store. That's super easy, based on a trustworthy collection, etc. In other words: my grandparents could do this on their own 😉

But on Firefox for Android, I have to:
1. Enable Developer Mode
2. Accept add-ons from a non-official store
3. Install a *monkey add-on
4. Install a Userscript

And I'm asking myself: why the heck do I have to do all these manual and pretty techy steps on Android? As a developer, I'm okay with that, but my grandparents definitely not 😁 Because basically I'm telling them all the time _Please install only from sources you know_, but they neither have an idea of non-official add-on stores nor a Userscript manager.

And all these steps wouldn't be necessary if the desktop add-on of PrivacyRedirect would be available for Android as well. So it boils down to: does Firefox reject PrivacyRedirect when proposing to add it to Firefox' curated list of Android add-ons? Or why are all these techy steps necessary?

**Comment by sjehuda** [comment]
I didn't know one needs to pass all these steps.
I guess they put these restrictions because they can't supervise Userscript files as they do with extensions.

**Comment by mu88** [comment]
But the Desktop add-on is no Userscript, it is an add-on. So back to the very first question: why not provide the Desktop add-on for Android?

**Comment by sjehuda** [comment]
No question, this addon is a worthwhile be available for Android too.

For the time being, I suggest you to install AdAway and set the proper set of redirection rules.  I think that won't require Developer mode.

Please let me ask you this question, how do you convince you loved ones (your grandparents) to install privacy-respecting software?

**Comment by mu88** [comment]
The grandparents were just a fictional example 😉 some are completely immune for these topics, but others can be convinced with a lot of explaining, showing bigger pictures, etc. But it's a tough topic, I agree

**Comment by sjehuda** [comment]
You can add a hosts file with rules to redirect to your desired
websites, with AdAway.
